samples: timer wakeup (#9)
NordicSemiconductor/npmx-zephyr: samples/timer_wakeup/README.rst

.. _timer_wake_up_sample:

TIMER wake-up
#############

.. contents::
   :local:
   :depth: 2

The TIMER wake-up sample demonstrates how to use PMIC TIMER and SHIPHLD with npmx drivers.

Requirements
************

The sample supports the following development kits:

.. table-from-sample-yaml::

The sample also requires an nPM1300 EK.

Overview
********

This sample uses the TIMER and SHIPHLD driver to do the following:

  * Configure SHIPHLD button.
  * Register handler to be called when the button is pressed.

>   * Configure TIMER in wake-up mode.

Author: Periphery

       - P1.03
       - P0.27
     * - VOUT2 & GND
       - External supply (P21)
       - External supply (P21)

#. Make the following connections on the nPM1300 EK:

   * Connect a USB power supply to the **J3** connector.
   * Connect a suitable battery to the **J2** connector.
   * On the **P18** pin header, connect **VOUT2** and **VDDIO** pins with a jumper.
   * On the **P2** pin header, connect **VBAT** and **VBATIN** pins with a jumper.
   * On the **P17** pin header, connect all LEDs with jumpers.
   * On the **P13** pin header, connect **RSET1** and **GND** pins with a jumper.
   * On the **P14** pin header, connect **RSET2** and **VSET2** pins with a jumper.

#. Make the following connections on the chosen DK:

   * Set the **SW9** nRF power source switch to **VDD**.
   * Set the **SW10** VEXT → nRF switch to **ON**.

Configuration
*************

|config|

Building and running
********************

.. |sample path| replace:: :file:`samples/timer_wakeup`

.. include:: /includes/build_and_run.txt

Testing
=======

|test_sample|

#. |connect_kit|
#. |connect_terminal|
#. Press SW1 button (SHPHLD) to enter the hibernate mode.
#. Wait for ~8 seconds until device wakes up.

> #. LEDs is used to indicate when the PMIC wakes up.

Author: The LED is on when PMIC is up.

.. _timer_wake_up_sample:

Wake-up Timer
#############

.. contents::
   :local:
   :depth: 2

> The Wake-up Timer sample demonstrates how to use the PMIC TIMER periphery and SHIPHLD button with npmx drivers.

Reviewer: ```suggestion
The Wake-up Timer sample demonstrates how to use the PMIC TIMER periphery and the SHIPHLD button with npmx drivers.
```

     * - GPIO0
       - P1.10
       - P1.10
     * - SDA
       - P1.02
       - P0.26
     * - SCL
       - P1.03
       - P0.27
     * - VOUT2 & GND
       - External supply (P21)
       - External supply (P21)

#. Make the following connections on the nPM1300 EK:

   * Connect a USB power supply to the **J3** connector.
   * Connect a suitable battery to the **J2** connector.
   * On the **P18** pin header, connect **VOUT2** and **VDDIO** pins with a jumper.
   * On the **P2** pin header, connect **VBAT** and **VBATIN** pins with a jumper.
   * On the **P17** pin header, connect all LEDs with jumpers.
   * On the **P13** pin header, connect **RSET1** and **GND** pins with a jumper.
   * On the **P14** pin header, connect **RSET2** and **VSET2** pins with a jumper.

#. Make the following connections on the chosen DK:

   * Set the **SW9** nRF power source switch to **VDD**.
   * Set the **SW10** VEXT → nRF switch to **ON**.

Configuration
*************

|config|

Building and running
********************

.. |sample path| replace:: :file:`samples/timer_wakeup`

.. include:: /includes/build_and_run.txt

Testing
=======

|test_sample|

#. |connect_kit|
#. |connect_terminal|

> #. Press the SW1 (SHPHLD) button to enter the hibernate mode.

Reviewer 2: `hibernate mode` or `hibernation mode`?

> #. Wait for ~8 seconds until device wakes up.

Reviewer: ```suggestion
#. Wait for ~8 seconds until the device wakes up.
```

> LEDs are used to indicate when the PMIC wakes up.

Reviewer: Here we should have a result of testing actions, so the sentence is in a correct place but we need to change its structure, for example:
```suggestion
LEDs will start blinking when the PMIC wakes up.
```

> LED is on when the PMIC wakes up.

Reviewer: ```suggestion
When the PMIC wakes up, the LED will turn on.
```
